meshinfo/aredn.py

```diff
@@ -424,82 +421,82 @@ def load_system_info(json_data: dict[str, Any], *, ip_address: str = "") -> Syst
-    ]
-
-    # create a dictionary with all the parameters due to the number
-    # and variance between API versions
-    data = {
-        "connection_ip": ip_address,
-        "node_name": json_data["node"].lower(),
-        "display_name": json_data["node"],
-        "api_version": json_data["api_version"],
-        "grid_square": json_data.get("grid_square", ""),
-        "latitude": float(json_data["lat"]) if json_data.get("lat") else None,
-        "longitude": float(json_data["lon"]) if json_data.get("lon") else None,
-        "interfaces": {iface.name: iface for iface in interfaces},
-        "services": [
-            Service.from_json(service_data)
-            for service_data in json_data.get("services_local", [])
-        ],
-        "services_json": json_data.get("services_local", []),
-        "source_json": json_data,
-    }
-
-    # generally newer versions add data in nested dictionaries
-    # sometimes that data was present at the root level in older versions
-
-    if "sysinfo" in json_data:
-        data["up_time"] = json_data["sysinfo"]["uptime"]
-        data["load_averages"] = [float(load) for load in json_data["sysinfo"]["loads"]]
-
-    if "meshrf" in json_data:
-        meshrf = json_data["meshrf"]
-        data["status"] = meshrf.get("status", "on")
-        #
-        data["ssid"] = meshrf.get("ssid", "")
-        data["channel"] = meshrf.get("channel", "")
-        data["channel_bandwidth"] = meshrf.get("chanbw", "")
-        data["frequency"] = meshrf.get("freq", "")
-    else:
-        data["ssid"] = json_data["ssid"]
-        data["channel"] = json_data["channel"]
-        data["channel_bandwidth"] = json_data["chanbw"]
-        data["status"] = "on"
-
-    if "node_details" in json_data:
-        details = json_data["node_details"]
-        data["description"] = html.unescape(details.get("description", ""))
-        data["firmware_version"] = details["firmware_version"]
-        data["firmware_manufacturer"] = details["firmware_mfg"]
-        data["model"] = details["model"]
-        data["board_id"] = details["board_id"]
-    else:
-        data["firmware_version"] = json_data["firmware_version"]
-        data["firmware_manufacturer"] = json_data["firmware_mfg"]
-        data["model"] = json_data["model"]
-        data["board_id"] = json_data["board_id"]
-
-    if "tunnels" in json_data:
-        tunnels = json_data["tunnels"]
-        data["active_tunnel_count"] = int(tunnels["active_tunnel_count"])
-    else:
-        data["active_tunnel_count"] = int(json_data["active_tunnel_count"])
-
-    if link_info := json_data.get("link_info"):
-        data["links"] = [
+    api_version = tuple(int(value) for value in json_data["api_version"].split("."))
+    if api_version < (1, 5):
+        # pre-1.5 the JSON was much flatter, so treat separately
+        logger.debug("Parsing legacy sysinfo.json", version=api_version)
+        return _load_legacy_system_info(json_data)
```

Comment:
Turns out that our mesh network only has a single node pre-1.5 (there's another one that reports that old, but it's not an AREDN node and thus is actually returning the newer format).

I think it makes more sense to just try the "modern" parser and then on exception try the legacy one.